File: AirHockeyServer/AirHockeyServer/Entities/MapEntity.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;

namespace AirHockeyServer.Entities
{
    ///////////////////////////////////////////////////////////////////////////////
    /// @file MapEntity.cs
    /// @author Michael Sghaier
    /// @date 2017-10-02
    /// @version 0.1
    ///
    /// Cette classe représente une carte de jeu
    ///////////////////////////////////////////////////////////////////////////////
    public class MapEntity : Entity
    {
        public int? Id { get; set; }
        public string Creator { get; set; }
        public string MapName { get; set; }
        public DateTime LastBackup { get; set; }
        public string Json { get; set; }
        public bool Private { get; set; }
        public string Password { get; set; }
        public int CurrentNumberOfPlayer { get; set; }
    }
}

File: AirHockeyLightClient/AirHockeyLightClient/MapEntity.swift
///////////////////////////////////////////////////////////////////////////////
/// @file MapEntity.swift
/// @author Mikael Ferland
/// @date 2017-10-22
/// @version 1
///
/// @addtogroup log3900 LOG3990
/// @{
///////////////////////////////////////////////////////////////////////////////

import RealmSwift

///////////////////////////////////////////////////////////////////////////
/// @class MapEntity
/// @brief Classe pour encapsuler une carte
///
/// @author Mikael Ferland
/// @date 2017-10-01
///////////////////////////////////////////////////////////////////////////
class MapEntity : Object {
    
    dynamic var id: String?
    dynamic var creator: String?
    dynamic var mapName: String?
    dynamic var lastBackup: Date?
    dynamic var json: String?
    let privacy = RealmOptional<Bool>()
    dynamic var password: String?
    let currentNumberOfPlayer = RealmOptional<Int>()
    
}

///////////////////////////////////////////////////////////////////////////////
/// @}
///////////////////////////////////////////////////////////////////////////////
